boskos/metrics/metrics.go

```diff
@@ -0,0 +1,171 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/test-infra/boskos/client"
+)
+
+type prometheusMetrics struct {
+	GceStats map[string]prometheus.Gauge
+	GkeStats map[string]prometheus.Gauge
+}
+
+var (
+	promMetrics = prometheusMetrics{
+		GceStats: map[string]prometheus.Gauge{
+			"free": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gce_project_free",
+				Help: "Number of free gce-project",
+			}),
+			"busy": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gce_project_busy",
+				Help: "Number of busy gce-project",
+			}),
+			"dirty": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gce_project_dirty",
+				Help: "Number of dirty gce-project",
+			}),
+			"cleaning": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gce_project_cleaning",
+				Help: "Number of cleaning gce-project",
+			}),
+		},
+		GkeStats: map[string]prometheus.Gauge{
+			"free": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_free",
+				Help: "Number of free gke-project",
+			}),
+			"busy": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_busy",
+				Help: "Number of busy gke-project",
+			}),
+			"dirty": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_dirty",
+				Help: "Number of dirty gke-project",
+			}),
+			"cleaning": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_cleaning",
+				Help: "Number of cleaning gke-project",
+			}),
+		},
+	}
+)
+
+func init() {
+	for _, gce := range promMetrics.GceStats {
+		prometheus.MustRegister(gce)
+	}
+
+	for _, gke := range promMetrics.GkeStats {
+		prometheus.MustRegister(gke)
+	}
+}
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	boskos := client.NewClient("Reaper", "http://boskos")
+	logrus.Infof("Initialzied boskos client!")
+
+	http.Handle("/prometheus", promhttp.Handler())
+	http.Handle("/", handleMetric(boskos))
+
+	go func() {
+		logTick := time.NewTicker(time.Minute).C
+		for {
+			select {
+			case <-logTick:
+				if err := update(boskos); err != nil {
+					logrus.WithError(err).Warning("[Boskos Metrics]Update failed!")
+				}
+			}
+		}
+	}()
+
+	logrus.Info("Start Service")
+	logrus.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
+}
+
+func update(boskos *client.Client) error {
+	if gce, err := boskos.Metric("gce-project"); err != nil {
+		return fmt.Errorf("fail to get metric for gce-project : %v", err)
+	} else {
```

Comment:
nit `else` is unnecessary here and in the conditional below since you return if the condition is true.